lld/ELF/Target.cpp

```diff
@@ -40,6 +41,12 @@ using namespace lld::elf;
 
 std::string elf::toStr(Ctx &ctx, RelType type) {
   StringRef s = getELFRelocationTypeName(ctx.arg.emachine, type);
+  if (ctx.arg.emachine == EM_RISCV && s == "Unknown") {
+    auto VendorString = getRISCVVendorString(type);
+    if (VendorString)
+      s = getRISCVVendorRelocationTypeName(type & ~INTERNAL_RISCV_VENDOR_MASK,
+                                           *VendorString);
+  }
```

Comment:
Do we want to do a fallback here for `R_RISCV_CUSTOM<n>` or not? Presumably we only get here when LLD managed some but not enough processing of the reloc, so maybe this is only going to be called for relocation numbers lld knows about?

Reply:
I'll add the fallback path. LLD should print *something* sane if run on object files containing vendor-relics it doesn't (yet) understand. If we don't include the relocation number, it makes it impossible for user to figure out what's going wrong.